src/rdf4cpp/rdf/Namespace.hpp

```diff
@@ -0,0 +1,87 @@
+#ifndef RDF4CPP_NAMESPACE_HPP
+#define RDF4CPP_NAMESPACE_HPP
+
+#include <rdf4cpp/rdf/IRI.hpp>
+#include <rdf4cpp/rdf/storage/util/robin-hood-hashing/robin_hood_hash.hpp>
+#include <rdf4cpp/rdf/storage/util/tsl/sparse_map.h>
+
+#include <string>
+#include <string_view>
+
+namespace rdf4cpp::rdf {
+
+/**
+ * Namespace provides a simple tool to create IRIs from a certain namespace and vocabulary. Each Namespace instance maintains an internal cache to save roundtrip to storage::node::NodeStorage.<br/>
+ * Example:
+ *  @code
+ *  Namespace ex("http://example.com/);
+ *  IRI me = ex + "me";
+ *  @endcode
+ */
+class Namespace {
+public:
+    using NodeStorage = storage::node::NodeStorage;
+
+protected:
+    /**
+     * The IRI string of the namespace.
+     */
+    std::string namespace_iri_;
+
+    /**
+     * NodeStorage from which IRI objects are created.
+     */
+    mutable NodeStorage node_storage_;
+
+    // TODO: a faster, less memory efficient map would be better.
+    /**
+     * Cache storing the <div>NodeBackendHandle</div> for prefixes. This saves roundtrips to NodeStorage.
+     */
+    storage::util::tsl::sparse_map<std::string, storage::node::identifier::NodeBackendHandle,
+                                   storage::util::robin_hood::hash<std::string>,
+                                   std::equal_to<std::string>>
+            cache_;
+
+public:
+    /**
+     * Namespace Constructor
+     * @param namespace_iri namespace IRI string. This will be used as prefix. IRI must not be encapsulated in <..>.
+     * @param node_storage where the IRIs will live
+     */
+    explicit Namespace(std::string_view namespace_iri,
+                       NodeStorage &node_storage = NodeStorage::default_instance());
+
+    /**
+     * @return IRI string of the Namespace
+     */
+    std::string const &names_space() const noexcept;
```

Comment:
Good point. What we loose, is that `auto x = rdfs.std::string{"debug: "} + x.name_space();` won't work anymore. That seems OK given that we do have a interface that is string implementation agnostic. 

name is a typo, should be `name_space` 👍🏻.

Reply:
I think the standard committee just forgot to provide an operator+ for string and string_view
because operator+= on string has a string_view overload; so you can emulate the missing operator+ by:

```c++
auto s = (std::string{"debug: "} += x.name_space());
```

which is kinda ugly but at least we don't loose that behaviour